### PR TITLE
machine driver update for UP2

### PR DESCRIPTION
**Pull request by xiulipan**
Only fix cpu_dai name for bxt_pcm512x machine driver

### CODE REVIEW

File: sound/soc/intel/boards/bxt_pcm512x.c [hunk 1]
```diff
@@ -106,9 +106,25 @@ static struct snd_soc_dai_link dailink[] = {
 	/* CODEC<->CODEC link */
 	/* back ends */
 	{
-		.name = "SSP5-Codec",
+		/* DMIC0 - Codec */
+		.name = "DMIC0",
 		.id = 0,
-		.cpu_dai_name = "sof-audio",
+		.init = NULL,
```
Comment (lgirdwood) [inline sound/soc/intel/boards/bxt_pcm512x.c]: Will be set to NULL by default.

File: sound/soc/intel/boards/bxt_pcm512x.c [hunk 2]
```diff
@@ -106,9 +106,25 @@ static struct snd_soc_dai_link dailink[] = {
...
+		.cpu_dai_name = "DMIC01 Pin",
+		.no_pcm = 1,
+		.platform_name = "sof-audio",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.codec_name = "snd-soc-dummy",
+		.dai_fmt = SND_SOC_DAIFMT_PDM,
+		.ignore_suspend = 1,
+		.be_hw_params_fixup = NULL,
+		.dpcm_playback = 0,
```
Comment (lgirdwood) [inline sound/soc/intel/boards/bxt_pcm512x.c]: 0 by default.

**Comment by xiulipan** [comment]
@plbossart 
The tplg file sof-apl-pcm512.tplg has already with this changes.
This is only effects UP2 platform now.
@lgirdwood 
I will update the patch.
Thanks

File: sound/soc/intel/boards/bxt_pcm512x.c [hunk 3]
```diff
@@ -106,9 +106,23 @@ static struct snd_soc_dai_link dailink[] = {
...
 		.id = 0,
-		.cpu_dai_name = "sof-audio",
+		.cpu_dai_name = "DMIC01 Pin",
+		.no_pcm = 1,
+		.platform_name = "sof-audio",
+		.codec_dai_name = "snd-soc-dummy-dai",
+		.codec_name = "snd-soc-dummy",
+		.dai_fmt = SND_SOC_DAIFMT_PDM,
+		.ignore_suspend = 1,
+		.be_hw_params_fixup = NULL,
```
Comment (plbossart) [inline sound/soc/intel/boards/bxt_pcm512x.c]: this is not needed.

**Comment by xiulipan** [comment]
Let me try to add the DMIC dai with some dynamic method. And try to use some all kinds of topology.

**Comment by xiulipan** [comment]
@plbossart 
Test with some simple method, but could not make run-time switching work.
Let merge the cpu dai name patch for the UP2 machine driver first
I will send a new RP for the DMIC support once I done.

**Comment by keyonjie** [comment]
@xiulipan you can swap the id for sof-apl-pcm512x.m4 if we don't plan to add dai_link for dmic to machine driver atm.

@plbossart the .cpu_dai_name = "sof-audio" should be changed to "SSP5 Pin", we don't have "sof-audio" dai anymore.

for the dmic BE dai_link, I think we can add it here(if we don't want to change id for SSP5, we can leave 0 for SSP5 and set id=1 for dmic BE, and change them to corresponding value in sof-apl-pcm512x.m4), that won't influence guys who don't have DMIC, they only need to comment out dmic part in sof-apl-pcm512x.m4 for their local use, or use tplg from topology/test/ folder directly.

**Comment by xiulipan** [comment]
@lgirdwood @plbossart @keyonjie 
Add DMIC into the id 1 for not influence the normal SSP test topology.
Also have a topology fix for test topology and sof-apl-pcm512x topology.
test with https://github.com/thesofproject/soft/pull/19

**Comment by plbossart** [comment]
I don't want to pull this for now. There is no rush to add DMICs for this platform, the only person who can test this on vacation for 2 more weeks and we will need to support SSP slave and HDMI support in the meantime - not to mention that we need to rework the machine driver to deal with the clock API as requested by Mark Brown.

**Comment by xiulipan** [comment]
@plbossart 
So what about a patch only for the cpu_dai_name and capture enable?
This will allow test tplg loaded on UP2 with loopback support.

**Pull request merged**